tests/unit/utils/validation-errors-test.js

```diff
@@ -57,3 +64,15 @@ test('#buildMessage builds a custom message if custom message is a function', fu
     'correct custom error message is returned'
   );
 });
+
+test('#buildMessage can return a raw data structure', function(assert) {
+  let originalConfig = get(config, 'changeset-validations');
+  set(config, 'changeset-validations', { rawOutput: true });
+  let result = buildMessage('firstName', { type: 'present', value: 'testValue', context: { foo: 'foo' }})
+  assert.ok(typeof result !== 'string', 'the return value is an object')
+  let { message, type, context: { description } } = result
+  assert.equal(message, "{description} can't be blank", 'default message is given')
+  assert.equal(description, 'First name', 'description is sent')
+  assert.equal(type, 'present')
```

Comment:
also worthwhile to assert value is in the result obj?

Reply:
I'm not sure what you suggest here: that **all** the values should be asserted or that I have asserted too generously :p ?

Reply:
Oh forgot backticks.  ‘value’ k-v exists in the ‘result’ as well, right?  Was thinking we could assert that as well.

Reply:
Absolutely. PR updated